src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1368,6 +1370,63 @@ TypeWithAnnotations getNotNullIfNotNullOutputType(TypeWithAnnotations outputType
         }
 
 #nullable enable
+        /// <summary>
+        /// Returns true if the method signature must match, with respect to scoped for ref safety,
+        /// in overrides, interface implementations, or delegate conversions.
+        /// </summary>
+        internal static bool RequiresValidScopedOverrideForRefSafety(MethodSymbol? method)
+        {
+            if (method is null)
+            {
+                return false;
+            }
+
+            var parameters = method.Parameters;
+
+            // https://github.com/dotnet/csharplang/blob/main/proposals/low-level-struct-improvements.md#scoped-mismatch
+            // The compiler will report a diagnostic for _unsafe scoped mismatches_ across overrides, interface implementations, and delegate conversions when:
+            // - The method returns a `ref struct` or returns a `ref` or `ref readonly`, or the method has a `ref` or `out` parameter of `ref struct` type, and
+            // ...
+            int nRefParametersRequired;
+            if (method.ReturnType.IsRefLikeType ||
+                (method.RefKind is RefKind.Ref or RefKind.RefReadOnly))
+            {
+                nRefParametersRequired = 1;
+            }
+            else if (parameters.Any(p => (p.RefKind is RefKind.Ref or RefKind.Out) && p.Type.IsRefLikeType))
+            {
+                nRefParametersRequired = 2; // including the parameter found above
```

Comment:
I wasn't sure if this will need to be modified to account for the upcoming `ReturnOnlyScope` change. Perhaps at that time it would be worth changing this to try an approach where we simulate the base calling into the derived and see if escape rules are satisfied.

```cs
class Base
{
    public virtual Span<byte> M(scoped ref byte input)
    {
        // if we could call Derived.M like this, would we satisfy escape rules?
        return Derived.M(ref input);
    }
}
class Derived : Base
{
    public override Span<byte> M(ref byte input)
    {
    }
}
```